src/v/archival/archival_policy.cc

```diff
@@ -12,6 +12,7 @@
 
 #include "archival/logger.h"
 #include "archival/segment_reupload.h"
+#include "config/configuration.h"
```

Comment:
>To encourage better compaction ratios in cloud storage, this updates the
archival policy to only reupload if a segment has been window compacted.

So I guess the idea here is that by waiting for window compaction and not just self-compaction that the ratios improve?

If we didn't wait, would we reupload after window compaction too, so that this optimization is a double win: less uploads (number of uploads) and less data uploaded (bytes)?

Reply:
>So I guess the idea here is that by waiting for window compaction and not just self-compaction that the ratios improve?

That's right.

>If we didn't wait, would we reupload after window compaction too, so that this optimization is a double win: less uploads (number of uploads) and less data uploaded (bytes)?

If we didn't wait for windowed compaction, we would have only uploaded after sliding window compaction, which doesn't guarantee window compaction has occurred. That is, we would have done a compacted reupload once, and it would be up to chance whether it's sliding window compacted or self compacted